jujugui/static/gui/src/app/app.js

```diff
@@ -1657,26 +1627,64 @@ YUI.add('juju-gui', function(Y) {
     },
 
     /**
+      Create the new socket url based on the socket template and the new
+      environment details.
+
+      @method createSocketURL
+      @param {String} server The state-server host address for the environment.
+      @param {String} port The state-server port for the environment.
+      @param {String} uuid The uuid for the environment.
+    */
+    createSocketURL: function(server, port, uuid) {
+      var currentHost = window.location.hostname,
+          currentPort = window.location.port;
+
+      var protocol = this.get('socket_protocol') || 'wss';
+      var baseUrl = protocol + '://' + currentHost;
+      if (currentPort !== '') {
+        baseUrl += ':' + currentPort;
+      }
+      baseUrl += '/ws';
+
+      if (!server || !port) {
+        var apiAddress = window.juju_config.apiAddress.replace('wss://', '');
```

Comment:
instead of grabbing this from window, `apiAddress` and `socketTemplate` below should be available as attributes on the app instance.